coprocess_python.go

```diff
@@ -295,20 +296,20 @@ func NewCoProcessDispatcher() (coprocess.Dispatcher, error) {
 		paths = append(paths, v)
 	}
 
-	PythonSetEnv(paths...)
-
-	PythonInit()
-	PythonLoadDispatcher()
-
-	dispatcher, err := PythonNewDispatcher(middlewarePath, eventHandlerPath, bundlePaths)
-
-	C.PyEval_ReleaseLock()
-
-	if err != nil {
-		log.WithFields(logrus.Fields{
-			"prefix": "coprocess",
-		}).Error(err)
-	}
-
+	go func() {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+		PythonSetEnv(paths...)
+		PythonInit()
+		PythonLoadDispatcher()
+		dispatcher, err = PythonNewDispatcher(middlewarePath, eventHandlerPath, bundlePaths)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"prefix": "coprocess",
+			}).Error(err)
+		}
+		pythonInitDone <- true
+	}()
+	<-pythonInitDone
```

Comment:
Err, I see that you are actually returning it, but by writing to the parent's error variable instead. This is a bit unintuitive, and will break easily with data races if we ever touch this code. Better to explicitly pass the error via the channel, returning nil if there was no error.